### PR TITLE
add smollm3 support

### PR DESCRIPTION
**Pull request by nmntz**
## Essential Elements of an Effective PR Description Checklist
- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.

## Purpose
Testing matching logits for SmolLM3 using transformers backend
Testing some vllm's features (TBA..)

-> Works with https://github.com/huggingface/transformers/pull/38755

### CODE REVIEW

**Comment by hmellor** [comment]
Thank you for the PR. I have benchmarked this dedicated implementation against the Transformers backend and the performance gap is <5%.

Given that we plan to close this gap with further improvements to the Transformers backend and the Transformers modelling code, I'm going to close this as superseded by https://github.com/vllm-project/vllm/pull/22665.
